services/apps/alcs/src/alcs/application/application-timeline/application-timeline.service.ts

```diff
@@ -0,0 +1,406 @@
+        isFulfilled: true,
+        fulfilledDate: null,
+      });
+    }
+
+    if (application.notificationSentDate) {
+      events.push({
+        htmlText: "'Ready for Review' Notification Sent to Applicant",
+        startDate:
+          application.notificationSentDate.getTime() +
+          SORTING_ORDER.READY_FOR_REVIEW_NOTIFICATION,
+        fulfilledDate: null,
+        isFulfilled: true,
+      });
+    }
+  }
+
+  private mapSubtaskToEvent(subtask: CardSubtask): TimelineEvent {
+    return {
+      htmlText: `${subtask.type.label} Subtask`,
+      fulfilledDate: subtask.completedAt?.getTime() ?? null,
+      startDate: subtask.createdAt.getTime(),
+      isFulfilled: !!subtask.completedAt,
+    };
+  }
+
+  private async addDecisionEvents(
+    application: Application,
+    events: TimelineEvent[],
+  ) {
+    const decisions = await this.applicationDecisionRepo.find({
+      where: {
+        applicationUuid: application.uuid,
+        isDraft: false,
+      },
+      order: {
+        date: 'DESC',
+      },
+    });
+
+    if (decisions.length > 0) {
+      const mappedApplications = await this.applicationService.mapToDtos([
+        application,
+      ]);
+      const mappedApplication = mappedApplications[0];
+
+      for (const [index, decision] of decisions.entries()) {
+        if (decision.auditDate) {
+          events.push({
+            htmlText: `Audited Decision #${decisions.length - index}`,
+            startDate:
+              decision.auditDate.getTime() + SORTING_ORDER.AUDITED_DECISION,
+            fulfilledDate: null,
+            isFulfilled: true,
+          });
+        }
+
+        if (decision.chairReviewDate) {
+          events.push({
+            htmlText: `Chair Reviewed Decision #${decisions.length - index}`,
+            startDate:
+              decision.chairReviewDate.getTime() +
+              SORTING_ORDER.CHAIR_REVIEW_DECISION,
+            fulfilledDate: null,
+            isFulfilled: true,
+          });
+        }
+
+        events.push({
+          htmlText: `Decision #${decisions.length - index} Made${
+            decisions.length - 1 === index
+              ? ` - Active Days: ${mappedApplication.activeDays}`
+              : ''
+          }`,
+          startDate: decision.date!.getTime() + SORTING_ORDER.DECISION_MADE,
+          fulfilledDate: null,
+          isFulfilled: true,
+        });
+      }
+    }
+  }
+
+  private async addReconsiderationEvents(
+    application: Application,
+    events: TimelineEvent[],
+  ) {
+    const reconsiderations = await this.applicationReconsiderationRepo.find({
+      where: {
+        applicationUuid: application.uuid,
+      },
+      relations: {
+        card: {
+          subtasks: {
+            type: true,
+          },
+        },
+      },
+      order: {
+        submittedDate: 'DESC',
+      },
+    });
+
+    for (const [index, reconsideration] of reconsiderations.entries()) {
+      if (reconsideration.type.code === '33.1') {
```

Comment:
there is an enum for reconsideration types: RECONSIDERATION_TYPE

Reply:
switched to enum!